azurelinuxagent/ga/update.py

```diff
@@ -457,7 +457,7 @@ def _ensure_readonly_files(self):
     def _ensure_cgroups_initialized(self):
         configurator = CGroupConfigurator.get_instance()
         configurator.create_agent_cgroups(track_cgroups=True)
-        configurator.cleanup_old_cgroups()
+        configurator.cleanup_legacy_cgroups()
```

Comment:
Does the order matter? Creating the new cgroup and then deleting the legacy cgroup? or should we be clearing legacy cgroup first before creating the new agent cgroup.

Reply:
Yes, cleanup may need to move the PID to the new cgroup, which needs to be created first (by create_agent_cgroups)